packages/slo-stat-bot/src/slo-logic.ts

```diff
@@ -0,0 +1,504 @@
+  const reqAssignee = slo.complianceSettings.requiresAssignee;
+  if (reqAssignee === true) {
+    const isAssigned = await getSLOStatus.isAssigned(responders, assignees);
+    if (!isAssigned) {
+      return false;
+    }
+  }
+
+  //Checking if issue is responded within response time
+  const responseTime = slo.complianceSettings.responseTime;
+  if (responseTime !== 0) {
+    const listIssueComments = await getSLOStatus.getIssueCommentsList(
+      github,
+      owner,
+      repo,
+      issueNumber
+    );
+    const isInResponseTime = await getSLOStatus.isInResponseTime(
+      responders,
+      listIssueComments,
+      slo.complianceSettings.responseTime,
+      issueUpdateTime
+    );
+    if (!isInResponseTime) {
+      return false;
+    }
+  }
+
+  return true;
+};
+
+getSLOStatus.isInResponseTime = async function isInResponseTime(
+  responders: Set<string>,
+  listIssueComments: IssuesListCommentsItem[] | null,
+  resolutionTime: string | number,
+  issueCreatedTime: string
+): Promise<boolean> {
+  if (!listIssueComments) {
+    return true; //If API call to list issue comments failed, does not attempt to label the issue
+  }
+  for (const comment of listIssueComments) {
+    if (responders.has(comment.user.login)) {
+      const isValidTime = await getSLOStatus.isInDuration(
+        resolutionTime,
+        issueCreatedTime,
+        comment.created_at
+      );
+      if (isValidTime) {
+        return true;
+      }
+    }
+  }
+  return false;
+};
+
+getSLOStatus.isAssigned = async function isAssigned(
+  responders: Set<string>,
+  assignees: IssueAssignees[]
+): Promise<boolean> {
+  for (const assignee of assignees) {
+    if (responders.has(assignee.login)) {
+      return true;
+    }
+  }
+  return false;
+};
+
+getSLOStatus.getIssueCommentsList = async function getIssueCommentsList(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  issueNumber: number
+): Promise<IssuesListCommentsItem[] | null> {
+  try {
+    const listComments = await github.issues.listComments({
+      owner,
+      repo,
+      issue_number: issueNumber,
+    });
+    return listComments.data;
+  } catch (err) {
+    console.error(
+      `Error in getting issue comments for number ${issueNumber}\n ${err.request}`
+    );
+    return null;
+  }
+};
+
+getSLOStatus.getResponders = async function getResponders(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  slo: SLORules
+): Promise<Set<string>> {
+  const responders: Set<string> = new Set();
+  //Getting list of owners from a uri-reference
+  let owners = slo.complianceSettings.responders?.owners;
+  if (owners) {
+    owners = await getSLOStatus.convertToArray(owners);
+    for (const ownerPath of owners) {
+      const content = await getSLOStatus.getFilePathContent(
+        github,
+        owner,
+        repo,
+        ownerPath
+      );
+      const users = content.match(/@([^\s]+)/g);
+      if (users) {
```

Comment:
neat trick I learned for Node: you can just do users?.forEach to test whether it exists!